### DISCUSSION TITLE
Show Component with a Multi Select Checkbox

### DISCUSSION THREAD

**Discussion by thering1975TG** [Budibase/budibase#13389] [Help] [answered]
HI

This is an odd one as it is something that can be achieved with basic form builders on various CMS's, yet i cannot fathom a simple way of doing this.

I want to keep the form clean, so i have some containers for each form element that includes for example a title, a dropdown select box and a small text field

For this example lets say i have a section called Jumpers a section called jackets and a section called trousers

I only want these containers to be visible if i select a checkbox which is at the top of the form

So for example, i have a multi select checkbox with the same options:

Jumpers - Jackets - Trousers

If i Click on one of these options then that relvant form container should become visible.

However using handlebars this does not appear possible 

I am using the following condition set on the contaioner

<img width="769" alt="conditions" src="https://github.com/Budibase/budibase/assets/87522339/e9a5cf2b-1687-4ebb-b007-42268829cfd4">

That works fine if i click the text box for jumpers the relevant section shows and becomes visible

**However**

It appears you can only set this condition once and cannot add another condition for a different section i presume due to the way it stores multiple select 

Ie if i click Jumpers and Jackets it stores as Jumpers,Jackets

That means no further conditions will ever match As lets say i set the Jacket section to be visible when selecting Jacket, but i already have Jumpers selected also, the condition will look for the answer Jacket, but will see Jumpers,Jacket.

This probably would have been solved if the condition section had a "Contains" option but it does not.

This means i would probably have to write a condition for every single possible combination of  options for each section.

Is there a way to achieve this or should i just ditch the idea all together

**Comment by mikesealey** [comment, marked as answer]
Hi @thering1975TG I've been looking into this, and you're right, there isn't a "Contains" option. I've created a [feature request](https://github.com/Budibase/budibase/issues/13545) on your behalf.

I've played around with a rough scenario in the meantime that may help you find a solution.

Instead of using check-boxes, I've added some buttons that `Update State`.
![image](https://github.com/Budibase/budibase/assets/39269380/00623b24-5990-4194-86a9-b6c11135d3aa)

 I've set the key to (eg) `jumpersVisible` and used some JavaScript to set the value `return !$("State.JumpersVisible")`. In this example, I'm using the exclamation mark to invert from the existing value from `true` to `false`, or from `false` to `true`. Each time the button is clicked, it will behave like a toggle.
![image](https://github.com/Budibase/budibase/assets/39269380/44cb8334-2ebe-4b8b-9f65-503a2231a4c5)

I've then set a visibility condition on the container that holds Jumpers
Show Component IF {{ State.JumpersVisible }} Equals Boolean True
![image](https://github.com/Budibase/budibase/assets/39269380/12b4e142-152c-449b-81ee-54a2e1076fe2)

An extra little bit can be added to the text of the button using some JavaScript and making use of the exclamation mark again.
If JumpersVisible is true, we can set the text to "Hide Jumpers" and if it's not-true we can set it to "Show Jumpers".
![image](https://github.com/Budibase/budibase/assets/39269380/8545c422-e846-42bc-82ff-8f1c16996f82)

Here's the export of the app I used as a sandbox - feel free to take a closer look. If you're not sure how to import this, please take a look at the documentation for [Importing an Application](https://docs.budibase.com/docs/export-and-import-apps#import-an-application) 

[13389-export-1713865579297.tar.gz](https://github.com/Budibase/budibase/files/15074829/13389-export-1713865579297.tar.gz)

I hope this helps you on your way. You can also access community support on the [Budibase Discord Server](https://discord.com/invite/ZepTmGbtfF).

**Comment by thering1975TG** [comment]
Hiya

Many thanks for this i have downloaded the export and will have a play around.

Appreciate you looking into it and raising a feature request

Many Thanks
